tests/vmi_networking_test.go

```diff
@@ -1075,17 +1076,25 @@ func NewRandomVMIWithInvalidNetworkInterface() *v1.VirtualMachineInstance {
 }
 
 func createExpectConnectToServer(serverIP string, tcpPort int, expectSuccess bool) []expect.Batcher {
-	expectResult := "1"
+	expectResult := console.ShellFail
 	if expectSuccess {
-		expectResult = "0"
+		expectResult = console.ShellSuccess
+	}
+
+	var clientCommand string
+
+	if netutils.IsIPv6String(serverIP) {
```

Comment:
I think it will be better to use the common code as a function but have 2 expecters exposed to the tests:
- createExpectConnectToHTTPServer
- createExpectConnectToTCPServer

Reply:
Can you please explain why? This method is called from `verifyClientServerConnectivity`. With your approach I will have to check the ipFamily before each call to the `createExpectConnectTo...` relevant method.
I think it complicates the code and the readability of the test.

Reply:
The upper layer is suppose to know what it tests (ipv4, ipv6) and how (tcp or http). Pushing this decision down the stack is requiring the implementation to take this "business" decision and know about things it should not care about.

As this is a localized implementation I will not insist on this. I hope to see more explicit approaches in general to reduce such conditions on lower level helpers.